### ISSUE TITLE
Investigate performance of video use case

### ISSUE THREAD

**Issue by emalysz** [mozilla-mobile/fenix#9341]
_edit (mcomella)_: I expanded the scope of the bug to be all video use cases.

One of our non-start up user flows (detailed in #7380) is the performance of video, and one component of that would be making videos full screen.

I have filed the following issues that impact the perceived performance:

- https://github.com/mozilla-mobile/fenix/issues/9339: The bottom nav bar has an obvious flicker when moving to full screen.
- https://github.com/mozilla-mobile/fenix/issues/9340: The entire UI will flicker if the link automatically directs to a full screen video

**Comment by mcomella** [comment]
On my Pixel 2, I aggressively seeked, rotated the device, and play/paused and **was unable to detect any performance issues in fullscreen mode and non-fullscreen mode.** This is a big improvement over my past experiences where seeking would take a very long time.

However, I did find two additional bugs that are out of scope for the perf group:
- https://github.com/mozilla-mobile/fenix/issues/9939 video controls are tiny on non-mobile pages
- https://github.com/mozilla-mobile/fenix/issues/9940 Calling HTML5 APIs rapidly may permanently pause video

**Comment by mcomella** [comment]
I re-ran my test using 720p on YouTube (the default for me was 360p) and I was still unable to identify issues. I checked the NFR and the only RC there are dropped frames, which is covered by automated testing. I think we've covered this issue comprehensively.

Ideally I could have tested on a lower end device but I do not have one available to me.

@ecsmyth Can you confirm this is what you want to see out of investigations into non-startup use cases?

**Comment by mcomella** [comment]
I filed a separate issue for picture-in-picture performance analysis because it's a less used case than general video playback and it's not released yet: https://github.com/mozilla-mobile/fenix/issues/9943

**Comment by ecsmyth** [comment]
I'm checking with the Media team to confirm, but I think that we've hit the key points for Fenix responsiveness (as opposed to issues with the players themselves) related to video playback.

**Comment by mcomella** [comment]
As a sanity check, I took profiles during these fullscreen operations to see if we were doing anything silly. I found https://github.com/mozilla-mobile/fenix/issues/9972 but nothing else.

**Comment by mcomella** [comment]
Last action from me: in https://github.com/mozilla-mobile/fenix/pull/10068, I'm asking MarcLeclair to measure the duration it takes to enter and exit fullscreen mode on the Moto G5. Otherwise, still waiting on ecsmyth & the Media team.

**Comment by ecsmyth** [comment]
I recommend that we close this issue as `done`. Existing investigation found a number of functional issues, however, the responsiveness of this portion of he UX is predominately controlled by the MSE JS players.

**Comment by mcomella** [comment]
On the Galaxy S5 on youtube.com https://github.com/mozilla-mobile/fenix/pull/10068#issuecomment-618604527, we've found it takes (3 runs each):
- Enter fullscreen: 319ms, 380ms, and 638ms (rounded to Glean bucketing sizes)
- Exit fullscreen: 123ms, 615ms, 615ms to exit fullscreen (rounded to Glean bucketing sizes)

On the Moto G5+ (unknown run count, unknown web page), MarcLeclair gets https://github.com/mozilla-mobile/fenix/pull/10068#issuecomment-622423614:
- Enter fullscreen: 355ms (presumably rounded to Glean bucketing sizes)
- Exit fullscreen: ~1.4s (presumably rounded to Glean bucketing sizes)

> Existing investigation found a number of functional issues, however, the responsiveness of this portion of he UX is predominately controlled by the MSE JS players.

@ecsmyth Given that youtube.com is probably a fairly optimized site and the measured performance is significantly higher than our target performance, do we want to investigate performance optimizations to the underlying enter/exit fullscreen implementation?

**Comment by mcomella** [comment]
Let's reopen this if we decide to investigate the performance of video in the future.
